From 1ca7d4c8362a7dfedd86b086b463d5d636b49fe9 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Tue, 21 Oct 2025 22:55:34 -0400
Subject: [PATCH 5/7] Fedora: remove ttrpc dependency
Content-type: text/plain

Not in Fedora and not required if not building aa_token

Also remove protobuf, as it's only used via ttrpc.

Also remove protos crate dependency, as it depends on ttrpc.

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 Cargo.toml                                     | 3 ---
 attestation-agent/attestation-agent/Cargo.toml | 4 +---
 attestation-agent/kbs_protocol/Cargo.toml      | 5 +----
 3 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index aca486e..9c57378 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,7 +31,6 @@ log = "0.4.29"
 nix = "0.30"
 openssl = "0.10"
 prost = "0.14"
-protobuf = "3.7.2"
 rand = "0.9.2"
 reqwest = { version = "0.12", default-features = false }
 resource_uri = { path = "attestation-agent/deps/resource_uri" }
@@ -52,8 +51,6 @@ tonic = "0.14"
 tonic-build = "0.14"
 tonic-prost = "0.14"
 tonic-prost-build = "0.14"
-ttrpc = "0.9.0"
-ttrpc-codegen = "0.6.0"
 url = "2.5.4"
 uuid = "1"
 zeroize = "1.8.2"
diff --git a/attestation-agent/attestation-agent/Cargo.toml b/attestation-agent/attestation-agent/Cargo.toml
index 53d9543..a70aa15 100644
--- a/attestation-agent/attestation-agent/Cargo.toml
+++ b/attestation-agent/attestation-agent/Cargo.toml
@@ -33,7 +33,6 @@ hex.workspace = true
 kbs_protocol = { path = "../kbs_protocol", default-features = false, optional = true }
 kbs-types.workspace = true
 log.workspace = true
-protos = { path = "../../protos", default-features = false, optional = true }
 reqwest = { workspace = true, features = ["json"], optional = true }
 serde.workspace = true
 serde_json.workspace = true
@@ -108,5 +107,4 @@ openssl = ["kbs_protocol?/openssl"]
 
 # Binary RPC type
 bin = ["clap", "env_logger", "tokio/rt-multi-thread"]
-grpc = ["tokio/signal", "tonic", "protos/grpc"]
-ttrpc = ["dep:ttrpc", "tokio/signal", "protos/ttrpc"]
+grpc = ["tokio/signal", "tonic"]
diff --git a/attestation-agent/kbs_protocol/Cargo.toml b/attestation-agent/kbs_protocol/Cargo.toml
index f6b2b51..8139718 100644
--- a/attestation-agent/kbs_protocol/Cargo.toml
+++ b/attestation-agent/kbs_protocol/Cargo.toml
@@ -18,8 +18,6 @@ env_logger = { workspace = true, optional = true }
 jwt-simple.workspace = true
 kbs-types.workspace = true
 log.workspace = true
-protobuf = { workspace = true, optional = true }
-protos = { path = "../../protos", default-features = false, optional = true }
 reqwest = { workspace = true, default-features = false, features = ["cookies", "json"] }
 resource_uri.path = "../deps/resource_uri"
 serde.workspace = true
@@ -27,7 +25,6 @@ serde_json.workspace = true
 sha2.workspace = true
 thiserror.workspace = true
 tokio.workspace = true
-ttrpc = { workspace = true, optional = true }
 url.workspace = true
 zeroize.workspace = true
 
@@ -49,7 +46,7 @@ default = ["background_check", "passport", "openssl", "all-attesters", "bin"]
 
 passport = []
 # Allow to connect Attestation-Agent with TTRPC to get evidence, token, etc.
-aa_ttrpc = ["passport", "ttrpc/async", "protos/ttrpc"]
+aa_ttrpc = []
 
 background_check = ["tokio/time", "canon-json"]
 all-attesters = ["attester/all-attesters"]
-- 
2.53.0

